test/e2e/aks.go

```diff
@@ -184,20 +157,56 @@ func WaitForAKSSystemNodePoolMachinesToExist(ctx context.Context, input WaitForC
 	Eventually(func() bool {
 		opt1 := client.InNamespace(input.Namespace)
 		opt2 := client.MatchingLabels(map[string]string{
-			infrav1.LabelAgentPoolMode: string(infrav1.NodePoolModeSystem),
 			clusterv1.ClusterNameLabel: input.ClusterName,
 		})
+		opt3 := client.MatchingLabels(map[string]string{
+			infrav1.LabelAgentPoolMode: string(infrav1.NodePoolModeSystem),
+		})
+
+		var capzMPs []client.Object
 
 		ammpList := &infrav1.AzureManagedMachinePoolList{}
+		asommpList := &infrav1exp.AzureASOManagedMachinePoolList{}
```

Comment:
The gist here is to build up one big list of all the CAPZ ManagedMachinePools, regardless of the exact type whether it's an AzureManagedMachinePool or AzureASOManagedMachinePool. Then we can use their shared metadata structure to get the owning MachinePool which is what we're ultimately checking here.